### PR TITLE
Remove useless casts from code

### PR DESCRIPTION
**Pull request by Luni-4**

### CODE REVIEW

File: src/pred.rs [hunk 1]
```diff
@@ -38,13 +38,6 @@ pub fn derive_borders<T: num_traits::AsPrimitive<isize>>(
     _height: usize,
     stride: usize,
 ) -> (isize, isize, isize, isize, isize, isize) {
```
Comment (lu-zero) [inline src/pred.rs]: shouldn't be changed as well?

**Review by lu-zero — APPROVED** [review]
Looks good beside 1 nit.

File: src/pred.rs (hunk 1 above)
Comment (Luni-4) [reply src/pred.rs]: `plane` is an `isize` so it might have negative values. Or are we assuming that plane is always an `usize`?

Comment (lu-zero) [reply src/pred.rs]: plane is a slice of pixel values, and they are non-negative by definition.

Comment (Luni-4) [reply src/pred.rs]: That's good, I was deceived by the `isize` in the generic eh eh

Comment (lu-zero) [reply src/pred.rs]: We use those values to index within u8 so there is lots of room for improvement.

**Review by lu-zero — APPROVED** [review]

**Pull request merged**